langchain/agents/agent_toolkits/spark_sql/base.py

```diff
@@ -0,0 +1,110 @@
+"""SQL agent."""
+from typing import Any, Dict, List, Optional
+
+from langchain.agents.agent import AgentExecutor, AgentOutputParser
+from langchain.agents.agent_toolkits.spark_sql.prompt import (
+    FLEXIBLE_SQL_PREFIX,
+    FLEXIBLE_SQL_SUFFIX,
+    SQL_PREFIX,
+    SQL_SUFFIX,
+)
+from langchain.agents.agent_toolkits.spark_sql.spark_freestyle_parser import (
+    SparkSQLFreeStyleOutputParser,
+)
+from langchain.agents.agent_toolkits.spark_sql.toolkit import (
+    SparkFlexibleSQLToolkit,
+    SparkSQLToolkit,
+)
+from langchain.agents.mrkl.base import ZeroShotAgent
+from langchain.base_language import BaseLanguageModel
+from langchain.callbacks.base import BaseCallbackManager
+from langchain.chains.llm import LLMChain
+from langchain.chat_models import ChatOpenAI
+from langchain.memory import ConversationBufferMemory
+from langchain.spark_sql import SparkSQL
+
+
+def create_spark_analytics_agent_verified(
+    schema: str,
+    allow_freestyle: bool = False,
+) -> AgentExecutor:
+    """
+    Construct a spark analytics agent with ChatGPT-4,
+    which is verified during development.
+    """
+    spark_sql = SparkSQL(schema=schema)
+    llm = ChatOpenAI(temperature=0, model_name="gpt-4")
```

Comment:
Let's not bind gpt-4 here. Langchain is supposed to be general.

Reply:
@skcoirz I am reverting the related changes. You can still find the code in the git history.

Reply:
sure, I’m good with it. There is a tradeoff. We can chat more later.